Fail to edit question if option filed is empty

<!-- Issue title: [brief description of bug] -->

- **Environment**: 
master at [05be9b6c9fa21be9966aeab76af7c51dff6de533](https://github.com/TEAMMATES/teammates/commit/05be9b6c9fa21be9966aeab76af7c51dff6de533)

**Steps to reproduce**
1. Open the `Edit Feedback Session` page with an instructor account
1. Try to edit a question, with the option field "question description" empty
1. Click "save" button

![image](https://user-images.githubusercontent.com/25312978/52927536-097f3580-3376-11e9-8229-8b4f3e89d126.png)


**Expected behaviour**
The change is saved


**Actual behaviour**
Nothing happens


**Additional info**
The log info in server is the following

```
Feb 18, 2019 12:04:46 PM com.google.appengine.api.datastore.dev.LocalDatastoreService$11 run
INFO: Time to persist datastore: 32 ms
Feb 18, 2019 12:09:12 PM teammates.common.util.Logger info
INFO: Request received: [PUT] http://localhost:8080/webapi/question, Params: {"questionid": "aglub19hcHBfaWRyHQsSEEZlZWRiYWNrUXVlc3Rpb24YgICAgICAgAkM"}, Headers: {"Origin": "http://localhost:4200", "Cookie": "JSESSIONID=_ahsY6YcDVoSQVTh4g1SvGY_oA.node0; CSRF-TOKEN=5132A3E5206F02DDA3992B9C5590D296CCA02E1130EFE81C54EA1B0F0E7BBDD6; dev_appserver_login=test@example.com:true:185804764220139124118", "Accept": "application/json, text/plain, */*", "X-CSRF-TOKEN": "5132A3E5206F02DDA3992B9C5590D296CCA02E1130EFE81C54EA1B0F0E7BBDD6", "User-Agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/72.0.3626.96 Safari/537.36", "Referer": "http://localhost:4200/web/instructor/sessions/edit?courseid=abc.exa-demo&fsname=Session%20with%20different%20question%20types", "Host": "localhost:8080", "Accept-Language": "zh-CN,zh;q=0.9,en;q=0.8,zh-TW;q=0.7", "Content-Length": "574", "Content-Type": "application/json"}, Request ID: 5c6a2fe800084d0091032ad7
Feb 18, 2019 12:09:12 PM teammates.common.util.Logger severe
SEVERE: AssertionError caught by WebApiServlet: 
java.lang.AssertionError: Field to update is null
	at teammates.common.util.Assumption.fail(Assumption.java:60)
	at teammates.common.util.Assumption.assertNotNull(Assumption.java:122)
	at teammates.common.datatransfer.attributes.FeedbackQuestionAttributes$UpdateOptions$Builder.withQuestionDescription(FeedbackQuestionAttributes.java:871)
	at teammates.ui.webapi.action.SaveFeedbackQuestionAction.execute(SaveFeedbackQuestionAction.java:79)
	at teammates.ui.webapi.action.WebApiServlet.invokeServlet(WebApiServlet.java:73)
	at teammates.ui.webapi.action.WebApiServlet.doPut(WebApiServlet.java:50)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:710)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:867)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1623)
	at teammates.ui.webapi.action.OriginCheckFilter.doFilter(OriginCheckFilter.java:97)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1610)
	at com.googlecode.objectify.ObjectifyFilter.doFilter(ObjectifyFilter.java:48)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1610)
	at com.google.appengine.tools.development.ResponseRewriterFilter.doFilter(ResponseRewriterFilter.java:134)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1610)
	at com.google.appengine.tools.development.HeaderVerificationFilter.doFilter(HeaderVerificationFilter.java:34)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1610)
	at com.google.appengine.api.blobstore.dev.ServeBlobFilter.doFilter(ServeBlobFilter.java:63)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1610)
	at com.google.apphosting.utils.servlet.TransactionCleanupFilter.doFilter(TransactionCleanupFilter.java:48)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1610)
	at com.google.appengine.tools.development.jetty9.StaticFileFilter.doFilter(StaticFileFilter.java:123)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1610)
	at com.google.appengine.tools.development.DevAppServerModulesFilter.doDirectRequest(DevAppServerModulesFilter.java:366)
	at com.google.appengine.tools.development.DevAppServerModulesFilter.doDirectModuleRequest(DevAppServerModulesFilter.java:349)
	at com.google.appengine.tools.development.DevAppServerModulesFilter.doFilter(DevAppServerModulesFilter.java:116)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1610)
	at com.google.appengine.tools.development.DevAppServerRequestLogFilter.doFilter(DevAppServerRequestLogFilter.java:44)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1602)
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:540)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:146)
	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:524)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:257)
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1588)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:255)
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1345)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:203)
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:480)
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1557)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:201)
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1247)
	at com.google.appengine.tools.development.jetty9.DevAppEngineWebAppContext.doScope(DevAppEngineWebAppContext.java:94)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:144)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
	at com.google.appengine.tools.development.jetty9.JettyContainerService$ApiProxyHandler.handle(JettyContainerService.java:595)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
	at org.eclipse.jetty.server.Server.handle(Server.java:502)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:364)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:260)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:305)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:103)
	at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:118)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:333)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:310)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:168)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:126)
	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:366)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:765)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:683)
	at java.lang.Thread.run(Thread.java:748)

```

--- a/src/main/java/teammates/common/datatransfer/attributes/FeedbackQuestionAttributes.java
+++ b/src/main/java/teammates/common/datatransfer/attributes/FeedbackQuestionAttributes.java
@@ -868,6 +868,7 @@ public class FeedbackQuestionAttributes extends EntityAttributes<FeedbackQuestio
             }
 
             public FeedbackQuestionAttributes.UpdateOptions.Builder withQuestionDescription(String questionDescription) {
+                // questionDescription can be null
                 updateOptions.questionDescriptionOption = UpdateOption.of(questionDescription);
                 return this;
             }

--- a/src/main/java/teammates/common/datatransfer/attributes/FeedbackQuestionAttributes.java
+++ b/src/main/java/teammates/common/datatransfer/attributes/FeedbackQuestionAttributes.java
@@ -868,8 +868,6 @@ public class FeedbackQuestionAttributes extends EntityAttributes<FeedbackQuestio
             }
 
             public FeedbackQuestionAttributes.UpdateOptions.Builder withQuestionDescription(String questionDescription) {
-                Assumption.assertNotNull(Const.StatusCodes.UPDATE_OPTIONS_NULL_INPUT, questionDescription);
-
                 updateOptions.questionDescriptionOption = UpdateOption.of(questionDescription);
                 return this;
             }

--- a/src/test/java/teammates/test/cases/datatransfer/FeedbackQuestionAttributesTest.java
+++ b/src/test/java/teammates/test/cases/datatransfer/FeedbackQuestionAttributesTest.java
@@ -573,15 +573,51 @@ public class FeedbackQuestionAttributesTest extends BaseAttributesTest {
     }
 
     @Test
+    public void testUpdateOptionsBuilder_withNullDescriptionInput_shouldUpdateAttributeCorrectly() {
+        FeedbackQuestionAttributes.UpdateOptions updateOptions =
+                FeedbackQuestionAttributes.updateOptionsBuilder("questionId")
+                        .withQuestionDescription(null)
+                        .build();
+
+        FeedbackQuestionAttributes questionAttributes =
+                FeedbackQuestionAttributes.builder()
+                        .withCourseId("courseId")
+                        .withFeedbackSessionName("session")
+                        .withGiverType(FeedbackParticipantType.INSTRUCTORS)
+                        .withRecipientType(FeedbackParticipantType.SELF)
+                        .withNumOfEntitiesToGiveFeedbackTo(3)
+                        .withQuestionNumber(1)
+                        .withQuestionType(FeedbackQuestionType.TEXT)
+                        .withQuestionMetaData(new FeedbackTextQuestionDetails("question text"))
+                        .withShowGiverNameTo(new ArrayList<>())
+                        .withShowRecipientNameTo(new ArrayList<>())
+                        .withShowResponseTo(new ArrayList<>())
+                        .build();
+
+        questionAttributes.update(updateOptions);
+
+        assertNull(questionAttributes.getQuestionDescription());
+        assertEquals("courseId", questionAttributes.getCourseId());
+        assertEquals("session", questionAttributes.getFeedbackSessionName());
+        assertEquals(FeedbackQuestionType.TEXT, questionAttributes.getQuestionType());
+        assertEquals("question text", questionAttributes.getQuestionDetails().getQuestionText());
+        assertEquals(1, questionAttributes.getQuestionNumber());
+        assertEquals(FeedbackParticipantType.INSTRUCTORS, questionAttributes.getGiverType());
+        assertEquals(FeedbackParticipantType.SELF, questionAttributes.getRecipientType());
+        assertEquals(3, questionAttributes.getNumberOfEntitiesToGiveFeedbackTo());
+        assertEquals(Lists.newArrayList(), questionAttributes.getShowResponsesTo());
+        assertEquals(Lists.newArrayList(), questionAttributes.getShowGiverNameTo());
+        assertEquals(Lists.newArrayList(), questionAttributes.getShowRecipientNameTo());
+
+    }
+
+    @Test
     public void testUpdateOptionsBuilder_withNullInput_shouldFailWithAssertionError() {
         assertThrows(AssertionError.class, () ->
                 FeedbackQuestionAttributes.updateOptionsBuilder(null));
         assertThrows(AssertionError.class, () ->
                 FeedbackQuestionAttributes.updateOptionsBuilder("id")
                         .withQuestionDetails(null));
-        assertThrows(AssertionError.class, () ->
-                FeedbackQuestionAttributes.updateOptionsBuilder("id")
-                        .withQuestionDescription(null));
         assertThrows(AssertionError.class, () ->
                 FeedbackQuestionAttributes.updateOptionsBuilder("id")
                         .withGiverType(null));

